src/components/discover-sheet/ListsSection.js

```diff
@@ -0,0 +1,334 @@
+import Routes from '@rainbow-me/routes';
+import { ethereumUtils } from '@rainbow-me/utils';
+
+const COINGECKO_TRENDING_ENDPOINT =
+  'https://api.coingecko.com/api/v3/search/trending';
+
+const fetchTrendingAddresses = async () => {
+  const trendingAddresses = [];
+  try {
+    const coingeckoIds = await fetchCoingeckoIds();
+    const request = await fetch(COINGECKO_TRENDING_ENDPOINT);
+    const trending = await request.json();
+    const idsToLookUp = trending.coins.map(coin => coin.item.id);
+    keys(coingeckoIds).forEach(address => {
+      if (idsToLookUp.indexOf(coingeckoIds[address]) !== -1) {
+        trendingAddresses.push(address);
+      }
+    });
+    // eslint-disable-next-line no-empty
+  } catch (e) {}
+  return trendingAddresses;
+};
+
+const ListButton = styled(ButtonPressAnimation).attrs({
+  scaleTo: 0.96,
+})`
+  margin-right: 16px;
+  ${({ selected, theme: { colors } }) =>
+    selected
+      ? `
+        background-color: ${colors.alpha(colors.blueGreyDark, 0.06)};
+        border-radius: 12px;
+        height: 30px;
+        padding-horizontal: 8px;
+        padding-top: ${ios ? 6 : 4}px;
+      `
+      : `
+        padding-top: ${ios ? 6 : 4}px;
+      `}
+`;
+
+const ListName = styled(Text)`
+  margin-left: 3px;
+  margin-top: ${ios ? -4.5 : 0}px;
+`;
+
+// Update trending lists every 5 minutes
+const TRENDING_LIST_UPDATE_INTERVAL = 5 * 60 * 1000;
+
+const layouts = [
+  { length: 110, offset: 0 },
+  { length: 89, offset: 110 },
+  { length: 120, offset: 200 },
+  { length: 81, offset: 320 },
+  { length: 92, offset: 400 },
+];
+const getItemLayout = (_, index) => ({
+  index,
+  ...layouts[index],
+});
+
+export default function ListSection() {
+  const dispatch = useDispatch();
+  const { network, nativeCurrency } = useAccountSettings();
+  const { navigate } = useNavigation();
+  const {
+    favorites,
+    lists,
+    updateList,
+    ready,
+    selectedList,
+    setSelectedList,
+    clearList,
+  } = useUserLists();
+  const listRef = useRef(null);
+  const initialized = useRef(false);
+  const { allAssets } = useAccountAssets();
+  const { genericAssets } = useSelector(({ data: { genericAssets } }) => ({
+    genericAssets,
+  }));
+  const { colors } = useTheme();
+  const listData = useMemo(() => DefaultTokenLists[network], [network]);
+
+  const addressSocket = useSelector(
+    ({ explorer: { addressSocket } }) => addressSocket
+  );
+  useEffect(() => {
+    if (addressSocket !== null) {
+      Object.values(listData).forEach(({ tokens }) => {
+        dispatch(emitAssetRequest(tokens));
+        dispatch(emitChartsRequest(tokens));
+      });
+    }
+  }, [addressSocket, dispatch, listData]);
+
+  const trendingListHandler = useRef(null);
+
+  const updateTrendingList = useCallback(async () => {
+    const tokens = await fetchTrendingAddresses();
+    clearList('trending');
+
+    dispatch(emitAssetRequest(tokens));
+    dispatch(emitChartsRequest(tokens));
+    tokens.forEach(address => {
```

Comment:
Given that we know that this is an for all the same list, can we expose a method in the userLists redux that allows you to add an array of token addresses at once? Otherwise this is updating the redux state for every single token on trending